python/igraphalg.py

```diff
@@ -26,154 +23,133 @@ def get_flow(
+        mgp.Record(node=graph.get_vertex_by_id(node_id), rank=rank)
+        for node_id, rank in enumerate(ranks)
     ]
 
 
 @mgp.read_proc
 def all_simple_paths(
     ctx: mgp.ProcCtx,
-    source: mgp.Vertex,
-    target: mgp.Vertex,
+    v: mgp.Vertex,
+    to: mgp.Vertex,
     cutoff: int = -1,
 ) -> mgp.Record(paths=mgp.List[mgp.List[mgp.Vertex]]):
-    graph, id_mapping, inverted_id_mapping = create_igraph_from_ctx(ctx, directed=True)
-    paths = [
-        convert_vertex_ids_to_mgp_vertices(path, ctx, inverted_id_mapping)
-        for path in graph.get_all_simple_paths(
-            v=id_mapping[source.id], to=id_mapping[target.id], cutoff=cutoff
-        )
-    ]
+    graph = MemgraphIgraph(ctx=ctx, directed=True)
 
-    return mgp.Record(paths=paths)
+    return mgp.Record(paths=graph.get_all_simple_paths(v=v, to=to, cutoff=cutoff))
 
 
 @mgp.read_proc
 def min_cut(
     ctx: mgp.ProcCtx,
     source: mgp.Vertex,
     target: mgp.Vertex,
-    edge_property: mgp.Nullable[str] = None,
-) -> mgp.Record(partition_vertex_ids=List[List[mgp.Vertex]], value=float):
-    graph, id_mapping, inverted_id_mapping = create_igraph_from_ctx(ctx, directed=True)
-    mincut = graph.mincut(
-        source=id_mapping[source.id],
-        target=id_mapping[target.id],
-        capacity=edge_property,
-    )
+    capacity: mgp.Nullable[str] = None,
+) -> mgp.Record(partition_vertices=List[List[mgp.Vertex]], value=float):
+    graph = MemgraphIgraph(ctx=ctx, directed=True)
+
+    cut = graph.mincut(source=source, target=target, capacity=capacity)
 
     return mgp.Record(
-        partition_vertex_ids=[
-            convert_vertex_ids_to_mgp_vertices(partition, ctx, inverted_id_mapping)
-            for partition in mincut.partition
+        partition_vertices=[
+            graph.convert_vertex_ids_to_mgp_vertices(vertex_ids=partition)
+            for partition in cut.partition
         ],
-        value=mincut.value,
+        value=cut.value,
     )
 
 
 @mgp.read_proc
 def topological_sort(
-    ctx: mgp.ProcCtx,
+    ctx: mgp.ProcCtx, directed: bool = True, mode: str = "out"
 ) -> mgp.Record(nodes=mgp.Nullable[mgp.List[mgp.Vertex]]):
-    graph, _, inverted_id_mapping = create_igraph_from_ctx(ctx, directed=True)
-    sorted = graph.topological_sorting(mode="out")
+    graph = MemgraphIgraph(ctx=ctx, directed=directed)
+    sorted_nodes = graph.topological_sort(mode=mode)
 
     return mgp.Record(
-        nodes=convert_vertex_ids_to_mgp_vertices(sorted, ctx, inverted_id_mapping)
+        nodes=sorted_nodes,
     )
 
 
 @mgp.read_proc
 def community_leiden(
     ctx: mgp.ProcCtx,
-    edge_property: mgp.Nullable[str] = None,
     resolution_parameter: float = 0.6,
-    number_of_iterations: int = -1,
+    n_iterations: int = -1,
+    beta: float = 0.01,
+    weights: mgp.Nullable[str] = None,
+    objective_function="modularity",
+    directed: bool = False,
 ) -> mgp.Record(community_id=int, community_members=List[mgp.Vertex]):
```

Comment:
Make the arguments consistent with [igraph](https://igraph.org/python/api/latest/igraph.Graph.html#community_leiden).